components/brave_wallet/browser/solana_provider_impl.cc

```diff
@@ -212,7 +215,9 @@ void SolanaProviderImpl::SignTransaction(
       MakeOriginInfo(delegate_->GetOrigin()), -1, *account,
       mojom::TxDataUnion::NewSolanaTxData(tx->ToSolanaTxData()),
       mojom::ByteArrayStringUnion::NewBytes(std::move(msg_pair->second)),
-      mojom::CoinType::SOL);
+      mojom::CoinType::SOL,
+      json_rpc_service_->GetChainId(mojom::CoinType::SOL,
+                                    delegate_->GetOrigin()));
```

Comment:
GetChainId returns purely prefs based result, so need in dependency on json_rpc_service. Worth at least a TODO to fix it later.

Reply:
I will remove the dependency.

Reply:
fixed in d77cae9e22e7962c4b5dc007a6fd909861e3bcd7